README.md
---
layout: home
title: "splash-media - Media files for `splash`"
image: assets/favimage-840x472.jpg
---

# splash-media

Media files for `splash` (https://github.com/RockingShip/splash)

!!  
!! This repository contains about 320M of media files  
!!

### Welcome to the Wonderful World of splash video codec

*When illustrations are 193x larger than the code*

This project contains the media files for: [https://github.com/RockingShip/splash](https://github.com/RockingShip/splash)

## Manifest

  - [ade/select.html](https://rockingship.github.io/splash-media/ade/select.html)  
    `splash` codec demo: "ADE" PixelsPerFrame selector (contains flashing white lights).

  - [gta/select.html](https://rockingship.github.io/splash-media/gta/select.html)  
    `splash` codec demo: "GTA" PixelsPerFrame selector (contains reckless driving).

  - [ade/ade-sbs-1800x506.webp](https://rockingship.github.io/splash-media/ade/ade-sbs-1800x506.webp)  
    [ade/ade-sbs-820x236.webp](https://rockingship.github.io/splash-media/ade/ade-sbs-820x236.webp)  
    Side-By-Side video of above for settings `PPF`=100 and upscaled 90x50.

  - [gta/gta-sbs-1800x506.webp](https://rockingship.github.io/splash-media/gta/gta-sbs-1800x506.webp)  
    [gta/gta-sbs-820x236.webp](https://rockingship.github.io/splash-media/gta/gta-sbs-820x236.webp)  
    Side-By-Side video of above for settings `PPF`=100 and upscaled 90x50.

  - [ade/ade-border-1920x1080.mp4](https://rockingship.github.io/splash-media/ade/ade-border-1920x1080.mp4)  
    [ade/ade-border-840x472.webp](https://rockingship.github.io/splash-media/ade/ade-border-840x472.webp)  
    Video displaying how `splash` constructs a single frame of "ADE".  
    The border displays processed scan row/columns.  
    The number in the lower left corner is progress (ranging 0-1).

  - [starships-1212x510-OU-crf18.mp4](starships/starships-1212x510-OU-crf18.mp4)  
    Original video is 24fps and slowed down 6x.  
    Top: 4fps full frame with traditional scan-lines.  
    Bottom: 1/6 PixelsPerFrame with scan-rows/cols and paint-brush.

## Source code

This repository contains about 320M of media files.

Checkout the latest code:

```sh
  git clone https://github.com/RockingShip/splash-media.git
```

There are no releases due to their size.

## Versioning

Using [SemVer](http://semver.org/) for versioning. For the versions available, see the [tags on this repository](https://github.com/RockingShip/splash-media/tags).

## License

This project is licensed under the GNU General Public License v3 - see the [LICENSE.txt](LICENSE.txt) file for details.  
The gallery is MIT licensed.


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

Media repositories have no releases due to their size.

```
2021-05-08 23:14:00 Added: Starship demo.
2021-04-07 14:21:22 Added: ADE source video.
2021-04-07 14:01:03 Added: ade/gta videos.
2021-04-06 20:46:45 Extracted from `jsFractalZoom`.
```


gta/gen-gta.sh
#!/bin/bash

if [ -z "$FFMPEG" ]; then FFMPEG=ffmpeg; fi

# extract 20 seconds from original video
if [ ! -f "gta-orig.720p.mp4" ]; then
  $FFMPEG -loglevel warning -stats -i cVGRDjlmMFk.mp4 -ss 8.2 -t 20 gta-orig.720p.mp4
fi

# unpack original into separate frames
rm -rf gta-frames
mkdir -p gta-frames
$FFMPEG -loglevel warning -stats -i gta-orig.720p.mp4 -vf scale=900:506 gta-frames/gta-%03d.png -y

for PPF in 1 2 5 10 20 50 100 150 200 300 600 900 1200 1600 2000 2400; do
	echo gta-${PPF}-900x506.webp
	$FFMPEG -loglevel warning -stats -r 30 -i gta-frames/gta-%03d.png -c:v splash -ppf $PPF -ppk 2 splash-gta-${PPF}-900x506.mkv -y
	$FFMPEG -loglevel warning -stats -r 30 -i splash-gta-${PPF}-900x506.mkv -q:v 85 gta-${PPF}-900x506.webp -y
done

# Upscale 90x50
echo Upscale 90x50
$FFMPEG -loglevel warning -stats -r 30 -i splash-gta-1-900x506.mkv  -vf scale=90:50   -crf 18 gta-90x50.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i gta-90x50.mkv     -vf scale=420:236 -crf 18 gta-90x50-420x236.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i gta-90x50.mkv     -vf scale=420:236 -q:v 85 gta-90x50-420x236.webp -y
$FFMPEG -loglevel warning -stats -r 30 -i gta-90x50.mkv     -vf scale=900:506 -crf 18 gta-90x50-900x506.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i gta-90x50.mkv     -vf scale=900:506 -q:v 85 gta-90x50-900x506.webp -y

# Upscale 285x160 to 420x236
echo Upscale 285x160
$FFMPEG -loglevel warning -stats -r 30 -i splash-gta-1-900x506.mkv  -vf scale=285:160 -crf 18 gta-285x160.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i gta-285x160.mkv   -vf scale=420:236 -crf 18 gta-285x160-420x236.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i gta-285x160.mkv   -vf scale=420:236 -q:v 85 gta-285x160-420x236.webp -y
$FFMPEG -loglevel warning -stats -r 30 -i gta-285x160.mkv   -vf scale=900:506 -crf 18 gta-285x160-900x506.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i gta-285x160.mkv   -vf scale=900:506 -q:v 85 gta-285x160-900x506.webp -y

# scale PPF=100 to 420x236
echo PPF=100
$FFMPEG -loglevel warning -stats -r 30 -i splash-gta-100-900x506.mkv  -vf scale=420:236 -crf 18 gta-100-420x236.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i splash-gta-100-900x506.mkv  -vf scale=900:506 -crf 18 gta-100-900x506.mkv -y

# Create gta side-by-size
echo SBS
$FFMPEG -loglevel warning -stats -i gta-100-420x236.mkv -i gta-90x50-420x236.mkv -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -q:v 85 gta-sbs-820x236.webp -y
$FFMPEG -loglevel warning -stats -i gta-100-900x506.mkv -i gta-90x50-900x506.mkv -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -q:v 85 gta-sbs-1800x506.webp -y

# preview
$FFMPEG -loglevel warning -stats -i gta-frames/gta-001.png -q 0 gta-900x506.jpg -y
$FFMPEG -loglevel warning -stats -i gta-100-420x236.mkv -i gta-90x50-420x236.mkv -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -ss 2 -frames:v 1 -q 0 gta-sbs-820x236.jpg -y
$FFMPEG -loglevel warning -stats -i gta-100-900x506.mkv -i gta-90x50-900x506.mkv -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -ss 2 -frames:v 1 -q 0 gta-sbs-1800x506.jpg -y

ffmpeg -loglevel warning -stats -i trek-20ppf.mp4 -i trek-6fps.mp4 -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -crf 18 -profile:v baseline -level 3.0 -movflags +faststart -pix_fmt yuv420p gta-sbs-1800x506.mp4 -y


ade/gen-ade.sh
#!/bin/bash

if [ -z "$FFMPEG" ]; then FFMPEG=ffmpeg; fi

# extract 20 seconds from original video
if [ ! -f "ade-orig.720p.mp4" ]; then
  $FFMPEG -loglevel warning -stats -i ade-20191018_025041.mp4 -t 10 -vf scale=1280:720 ade-orig.720p.mp4
fi

# unpack original into separate frames
rm -rf ade-frames
mkdir -p ade-frames
$FFMPEG -loglevel warning -stats -i ade-orig.720p.mp4 -vf scale=900:506 ade-frames/ade-%03d.png -y

for PPF in 1 2 5 10 20 50 100 150 200 300 600 900 1200 1600 2000 2400; do
	echo ade-${PPF}-900x506.webp
	$FFMPEG -loglevel warning -stats -r 30 -i ade-frames/ade-%03d.png -c:v splash -ppf $PPF -ppk 2 splash-ade-${PPF}-900x506.mkv -y
	$FFMPEG -loglevel warning -stats -r 30 -i splash-ade-${PPF}-900x506.mkv -q:v 85 ade-${PPF}-900x506.webp -y
done

# Upscale 90x50
echo Upscale 90x50
$FFMPEG -loglevel warning -stats -r 30 -i splash-ade-1-900x506.mkv  -vf scale=90:50   -crf 18 ade-90x50.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i ade-90x50.mkv     -vf scale=420:236 -crf 18 ade-90x50-420x236.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i ade-90x50.mkv     -vf scale=420:236 -q:v 85 ade-90x50-420x236.webp -y
$FFMPEG -loglevel warning -stats -r 30 -i ade-90x50.mkv     -vf scale=900:506 -crf 18 ade-90x50-900x506.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i ade-90x50.mkv     -vf scale=900:506 -q:v 85 ade-90x50-900x506.webp -y

# Upscale 285x160 to 420x236
echo Upscale 285x160
$FFMPEG -loglevel warning -stats -r 30 -i splash-ade-1-900x506.mkv  -vf scale=285:160 -crf 18 ade-285x160.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i ade-285x160.mkv   -vf scale=420:236 -crf 18 ade-285x160-420x236.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i ade-285x160.mkv   -vf scale=420:236 -q:v 85 ade-285x160-420x236.webp -y
$FFMPEG -loglevel warning -stats -r 30 -i ade-285x160.mkv   -vf scale=900:506 -crf 18 ade-285x160-900x506.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i ade-285x160.mkv   -vf scale=900:506 -q:v 85 ade-285x160-900x506.webp -y

# scale PPF=100 to 420x236
echo PPF=100
$FFMPEG -loglevel warning -stats -r 30 -i splash-ade-100-900x506.mkv  -vf scale=420:236 -crf 18 ade-100-420x236.mkv -y
$FFMPEG -loglevel warning -stats -r 30 -i splash-ade-100-900x506.mkv  -vf scale=900:506 -crf 18 ade-100-900x506.mkv -y

# Create ade side-by-size
echo SBS
$FFMPEG -loglevel warning -stats -i ade-100-420x236.mkv -i ade-90x50-420x236.mkv -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -q:v 85 ade-sbs-820x236.webp -y
$FFMPEG -loglevel warning -stats -i ade-100-900x506.mkv -i ade-90x50-900x506.mkv -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -q:v 85 ade-sbs-1800x506.webp -y
$FFMPEG -loglevel warning -stats -i ade-100-900x506.mkv -i ade-90x50-900x506.mkv -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -crf 18 -profile:v baseline -level 3.0 -movflags +faststart -pix_fmt yuv420p ade-sbs-1800x506.mp4 -y

# preview
$FFMPEG -loglevel warning -stats -i ade-frames/ade-001.png -q 0 ade-900x506.jpg -y
$FFMPEG -loglevel warning -stats -i ade-100-420x236.mkv -i ade-90x50-420x236.mkv -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -ss 2 -frames:v 1 -q 0 ade-sbs-820x236.jpg -y
$FFMPEG -loglevel warning -stats -i ade-100-900x506.mkv -i ade-90x50-900x506.mkv -filter_complex "[0:v][1:v]hstack=inputs=2[v]" -map "[v]" -ss 2 -frames:v 1 -q 0 ade-sbs-1800x506.jpg -y

#
# $FFMPEG -loglevel warning -stats -r:v 240/1 -i ade-border-1920x1080.mp4 -vf scale=840:472 -r 12/1 ade-border-840x472.webp
